tests/v2/validation/fleet/provisioning/airgap/airgap.go

```diff
@@ -0,0 +1,180 @@
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	return updateLocalFleetRepo(bastionUser, bastionIP, key)
+
+}
+
+// updateLocalFleetRepo clones fleet-examples repo onto the bastion node via ssh
+func updateLocalFleetRepo(bastionUser, bastionIP string, privateKey ssh.Signer) (string, error) {
+	config := &ssh.ClientConfig{
+		User: bastionUser,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(privateKey),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	conn, err := ssh.Dial(tcp, bastionIP+sshPort, config)
+	if err != nil {
+		return "", err
+	}
+	defer conn.Close()
+
+	// install git if it isn't on the bastion already
+	err = executeCommand(conn, "git version")
+	if err != nil {
+		gitSession, err := conn.NewSession()
+		if err != nil {
+			return "", err
+		}
+		defer gitSession.Close()
+
+		osOutput, err := gitSession.CombinedOutput("hostnamectl")
+		if err != nil {
+			return "", err
+		}
+		osString := string(osOutput)
+
+		if strings.Contains(osString, "Ubuntu") {
+			err = executeCommand(conn, "sudo apt-get update && sudo apt-get install -y git")
+		} else if strings.Contains(osString, "SUSE") {
+			err = executeCommand(conn, "sudo zypper refresh && sudo zypper install git -y")
+		}
+		if err != nil {
+			return "", err
+		}
+	}
+
+	err = executeCommand(conn, "cd ~/ && git clone --mirror "+rancherGithubURL+fleetExampleFolderName)
+	if err != nil {
+		if !strings.Contains(err.Error(), "128") {
+			return "", err
+		}
+	}
+
+	err = executeCommand(conn, "sudo git config --system --add safe.directory '*'")
+	if err != nil {
+		return "", err
+	}
+
+	session, err := conn.NewSession()
+	if err != nil {
+		return "", err
+	}
+	defer session.Close()
+
+	internalByteIP, err := session.CombinedOutput("hostname -I | awk '{print $1}'")
+
+	// output ends in a newline, which we don't need
+	return string(internalByteIP)[:len(internalByteIP)-1], err
+}
+
+// executeCommand executes a command via ssh. Each executed command needs its own session. See ssh package for more details.
+func executeCommand(client *ssh.Client, command string) error {
+	session, err := client.NewSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	session.Stdout = os.Stdout
+	session.Stderr = os.Stderr
+
+	return session.Run(command)
+}
+
+// createSSHSecret is a helper to create a secret using wrangler client for ssh in fleet-default
+func createSSHSecret(client *rancher.Client, data map[string][]byte) (*corev1.Secret, error) {
+	var err error
+
+	ctx := client.WranglerContext
+
+	secretName := namegenerator.AppendRandomString("fleet-ssh")
+	secretTemplate := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: fleetDefaultNS,
+		},
+		Data: data,
+		Type: corev1.SecretTypeSSHAuth,
+	}
+
+	createdSecret, err := ctx.Core.Secret().Create(&secretTemplate)
+
+	return createdSecret, err
+}
+
+// createFleetSSHSecret creates an sshSecret in fleet-default namespace for use with fleet resources
```

Comment:
is this going to be on the local cluster or the downstream cluster could you clarify?

Reply:
this is specific to creating in the fleet-default namespace on the local cluster.